server/channels/app/imaging/preview_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package imaging
+
+import (
+	"image"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerateThumbnail(t *testing.T) {
+	tcs := []struct {
+		name           string
+		inputImg       image.Image
+		targetWidth    int
+		targetHeight   int
+		expectedWidth  int
+		expectedHeight int
+	}{
+		{
+			name:         "empty image",
+			inputImg:     image.NewRGBA(image.Rect(0, 0, 0, 0)),
+			targetWidth:  120,
+			targetHeight: 100,
+		},
+		{
+			name:           "both dimensions lower than targets",
+			inputImg:       image.NewRGBA(image.Rect(0, 0, 100, 50)),
+			targetWidth:    120,
+			targetHeight:   100,
+			expectedWidth:  120,
+			expectedHeight: 60,
+		},
```

Comment:
This is the case I mean: do we need to enlarge the image from 100x50 to 120x60?

Reply:
@agarciamontoro I think upscaling is fine in this case since we are providing target dimensions, not merely constraints.